src/structured_data.rs

```diff
@@ -96,6 +96,29 @@ impl StructuredData {
         DataIdentifier::Structured(self.name, self.type_tag)
     }
 
+    /// Delete self
+    pub fn delete_self_against_successor(&mut self,
+                                         other: &StructuredData)
+                                         -> Result<(), RoutingError> {
+        match self.validate_self_against_successor(other) {
```

Comment:
An attacker being a client's accessing point do have the capbility to replace the Post with Delete.